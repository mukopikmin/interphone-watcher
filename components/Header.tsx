import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import AppBar from '@material-ui/core/AppBar'
import Toolbar from '@material-ui/core/Toolbar'
import Button from '@material-ui/core/Button'
import Link from 'next/link'
import Container from '@material-ui/core/Container'

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    flexGrow: 1,
  },
}))

export default function ButtonAppBar() {
  const classes = useStyles()

  return (
    <div className={classes.root}>
      <AppBar position="static">
        <Toolbar>
          <Container maxWidth="md">
            <Link href={'/'} passHref>
              <Button color="inherit" component="a">
                Interphone Watcher
              </Button>
            </Link>
          </Container>
        </Toolbar>
      </AppBar>
    </div>
  )
}
