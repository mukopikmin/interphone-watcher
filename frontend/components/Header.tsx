import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import AppBar from '@material-ui/core/AppBar'
import Toolbar from '@material-ui/core/Toolbar'
import Button from '@material-ui/core/Button'
import Link from 'next/link'

const useStyles = makeStyles(() => ({
  root: {
    flexGrow: 1,
    marginBottom: 70,
  },
  logo: {
    height: 36,
    verticalAlign: 'middle',
  },
}))

const Header: React.FC = () => {
  const classes = useStyles()

  return (
    <div className={classes.root}>
      <AppBar position="fixed">
        <Toolbar variant="dense">
          <span>
            <Link href={'/'}>
              <img
                className={classes.logo}
                src="/static/icons/icon-192x192.png"
                alt=""
              />
            </Link>
          </span>
          <Link href={'/'} passHref>
            <Button color="inherit" component="a">
              House Dashboard
            </Button>
          </Link>
        </Toolbar>
      </AppBar>
    </div>
  )
}

export default Header
