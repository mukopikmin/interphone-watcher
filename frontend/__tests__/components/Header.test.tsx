import React from 'react'
import { cleanup, render, screen } from '@testing-library/react'
import Header from '@/components/Header'

jest.mock('next/link', () => {
  const Link = ({
    href,
    children,
  }: {
    href: string
    children: string
  }): JSX.Element => <a href={href}>{children}</a>

  return Link
})

describe('Header', () => {
  afterEach(cleanup)
  beforeEach(() => render(<Header />))

  it('renders component as expected', () => {
    const button = screen.getByText('House Dashboard')
    const logo = screen.getByTestId('logo')

    expect(button).toBeInTheDocument()
    expect(logo).toBeInTheDocument()
  })

  it('has link to top on logo', () => {
    const button = screen.getByText('House Dashboard')

    expect(button.closest('a')).toHaveAttribute('href', '/')
  })
})
