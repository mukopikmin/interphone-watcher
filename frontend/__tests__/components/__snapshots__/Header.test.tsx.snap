// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders component as expected 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-dense MuiToolbar-gutters"
      >
        <a
          href="/"
        >
          <img
            alt=""
            class="makeStyles-logo-2"
            data-testid="logo"
            src="/static/icons/icon-192x192.png"
          />
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              House Dashboard
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
        </a>
      </div>
    </header>
  </div>
</div>
`;
